recipes/scip/all/conanfile.py

```diff
@@ -84,13 +80,21 @@ def config_options(self):
             del self.options.fPIC
 
     def requirements(self):
+        def _mapping_requires(dep, **kwargs):
+            required_version = self.conan_data["version_mappings"][self.version][dep]
+            self.requires(f"{dep}/{required_version}", **kwargs)
+
         if self.options.with_gmp:
             self.requires("gmp/6.3.0")
         if self.options.with_sym == "bliss":
             self.requires("bliss/0.77")
-        self.requires(f"soplex/{self.soplex_version_belonging_to_me[self.version]}")
+        _mapping_requires("soplex")
         self.requires("zlib/[>=1.2.11 <2]")
 
+    def config_options(self):
```

Comment:
this method is being redefined from elsewhere

Reply:
I don't understand this comment. What am I doing wrong?

Reply:
Sorry - it means that the recipe contains to definitions of the `config_options(self)` method - so one is overriding the other

Reply:
Whupps, I added a commit to merge both

Reply:
Thanks! :)